src/libraries/System.Private.CoreLib/src/System/Text/Unicode/Utf8Utility.Validation.cs

```diff
@@ -736,16 +732,20 @@ internal static unsafe partial class Utf8Utility
             return pInputBuffer;
         }
 
+        private static readonly Vector128<byte> s_NonAsciiBitMask128 = BitConverter.IsLittleEndian ?
```

Comment:
I believe this was @carlossanlop's first approach and this was causing a PNSE in some platforms because since it is a static field it is initialized at startup... but I can't remember exactly the error he was seeing.

Reply:
Also, this has less desirable performance characteristics compared to inline initialization.

Reply:
Using this approach, the whole thing goes away when building for non-arm64 because the linker will remove the single method that uses it and then remove the whole static field.

Other approaches would be:

- create it every time in the GetNonAsciiBytes method
- refactor the calling method so the loops are inside the ‘Arm64.IsSupported’ checks, which might need to dupe a bit of code.

Unfortunately I don’t have an Arm64 machine to perf test this myself.

Reply:
I was getting the PNSE on .NET Framework x86 CI legs for Utf8String.Experimental, since this file gets compiled into the Utf8String OOB package.

I refactored to use the 2nd approach above. It didn't turn out terrible.